inf-clojure.el

```diff
@@ -770,6 +806,64 @@ Return the number of nested sexp the point was over or after."
   (setq-local eldoc-documentation-function #'inf-clojure-eldoc)
   (apply #'eldoc-add-command inf-clojure-extra-eldoc-commands))
 
-(provide 'inf-clojure)
+;;;; Lumo
+;;;; ====
+
+(defgroup lumo nil
+  "Run an external Lumo process (REPL) in an Emacs buffer."
+  :group 'inf-clojure)
 
+(defcustom inf-clojure-lumo-command "lumo"
+  "The command used to launch lumo."
+  :type 'string
+  :group 'lumo)
+
+(defcustom inf-clojure-lumo-args "-d"
+  "The command arguments used to launch lumo."
+  :type 'string
+  :group 'lumo)
+
+;; AR - TODO Alternatively you can specify a command string that will be called,
+;; which should return a string.
+(defcustom inf-clojure-lumo-classpath-generator "cp"
+  "The file used to create the classpath string.
+The classpath string has to be a \":\" separated list of dir and
+files."
+  :type 'string
+  :group 'lumo)
+
+;; AR - not used but left here because it is a possible sanity check
+(defun inf-clojure--lumo-mode-p ()
+  "Return true if the lumo is the target REPL."
+  (comint-send-string (inf-clojure-proc) "(js/global.hasOwnProperty \"$$LUMO_GLOBALS\")"))
+
+(defun inf-clojure--lumo-program ()
+  "Return inf-clojure-program for lumo."
+  (concat inf-clojure-lumo-command
+          " "
+          (when (not (inf-clojure--empty-string-p inf-clojure-lumo-classpath-generator))
+            (concat inf-clojure-lumo-args " "))
+          "-c " (inf-clojure--read-classpath inf-clojure-lumo-classpath-generator)))
+
+(defun inf-clojure--flavor-lumo-setup ()
+  "Setup defcustoms for the Lumo flavor."
+  ;; The defcustoms for the following are already ok:
+  ;;  * inf-clojure-set-ns-command
+  ;;  * inf-clojure-macroexpand-command
+  ;;  * inf-clojure-macroexpand-1-command
+  ;;
+  ;; https://github.com/anmonteiro/lumo/issues/84
+  ;; (setq inf-clojure-var-source-command "(lumo.repl/source %s)")
+  ;; https://github.com/anmonteiro/lumo/issues/87
+  ;; (setq inf-clojure-ns-vars-command "(lumo.repl/dir %s)")
+  ;; https://github.com/anmonteiro/lumo/issues/86
+  ;; (setq inf-clojure-var-apropos-command "(lumo.repl/apropos %s)")
+
+  ;; Uncomment after https://github.com/anmonteiro/lumo/pull/88
+  ;; (setq inf-clojure-arglist-command "(lumo.repl/get-arglists \"%s\")")
+  (setq inf-clojure-var-doc-command "(lumo.repl/doc %s)")
```

Comment:
You should be setting and checking only buffer-locals. The defcustoms should simply exist as the values of the different flavours. This means you'll have one doc buffer-local and two defcustoms - one for clojure and one for lumo. Alternatively you can just have a buffer-local and hardcode the defcustom values, but this will make the code less flexible, so I wouldn't recommend it.